lib/factory.rb

```diff
@@ -14,3 +13,86 @@
 # - to_a
 # - values_at
 # - ==, eql?
+class Factory
+  class << self
+    def new(*args, &block)
+      const_set(args.shift.capitalize, class_new(*args, &block)) if args.first.is_a?(String)
+      class_new(*args, &block)
+    end
+
+    def class_new(*args, &block)
+      Class.new do
+        attr_accessor(*args)
+
+        def initialize(*vars)
+          raise ArgumentError, "Expected #{args.count} argument(s)" if args.count != vars.count
```

Comment:
`args` and `vars` naming looks a little strange. Can we change it? Maybe `class_args` and `instance_args`?